### PR TITLE
Created getAllCourseProgram api endpoint

### LINKED ISSUE #176: getAllCoursePrograms
- Endpoint URI: */course-programs/faculty*
- Http Method: *GET*

- Authorization
    - Members

- API Logic
    - get All Course Programs

- Progress
    - [ ] development
    - [ ] integration Test

### PR DESCRIPTION
**Pull request by junzai97**
Resolved #176 and #178

### CODE REVIEW

File: src/main/java/com/thirdcc/webapp/repository/CourseProgramRepository.java [hunk 1]
```diff
@@ -7,6 +7,8 @@
 import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
 import org.springframework.stereotype.Repository;
 
+import java.util.List;
+
```
Comment (HowardYaw) [inline src/main/java/com/thirdcc/webapp/repository/CourseProgramRepository.java]: why import this?

File: src/main/java/com/thirdcc/webapp/web/rest/CourseProgramResource.java [hunk 2]
```diff
@@ -50,4 +50,18 @@ public CourseProgramResource(CourseProgramService courseProgramService) {
...
+     *
+     * @return the {@link ResponseEntity} with status {@code 200 (OK)} and the list of faculties in body.
+     */
+    @GetMapping("/course-programs/faculty")
+    @PreAuthorize("isAuthenticated()")
+    public ResponseEntity<List<CourseProgram>> getAllCoursePrograms() {
+        log.debug("REST request to get all Course Programs");
+        List<CourseProgram> entityList = courseProgramService.findAll();
+        return ResponseEntity.ok().body(entityList);
+    }
```
Comment (HowardYaw) [inline src/main/java/com/thirdcc/webapp/web/rest/CourseProgramResource.java]: why your java doc and the API URL is not tally?
implementation is doing get all
but java doc seems like you want to find based on faculty id

File: src/main/java/com/thirdcc/webapp/repository/CourseProgramRepository.java (hunk 1 above)
Comment (junzai97) [reply src/main/java/com/thirdcc/webapp/repository/CourseProgramRepository.java]: I think I have added in accidently. Resolved in 3d3667fc1d901fb47691745754896ff1ea423842

File: src/main/java/com/thirdcc/webapp/web/rest/CourseProgramResource.java (hunk 2 above)
Comment (junzai97) [reply src/main/java/com/thirdcc/webapp/web/rest/CourseProgramResource.java]: Resolved in 3d3667fc1d901fb47691745754896ff1ea423842

**Review by HowardYaw — APPROVED** [review]

**Review by CarlosSia — APPROVED** [review]

**Pull request merged**